File: C-implementation/src/sir.c
#include "all.h"
#include "sir.h"

double dSdt(double* params, double* last) {
    return -1*params[b]*last[S]*last[I];
}

double dIdt(double* params, double* last) {
    return (params[b]*last[S]*last[I])-(params[k]*last[I]);
}

double dRdt(double* params, double* last) {
    return params[k]*last[I];
}

ode funcs[3] = {&dSdt, &dIdt, &dRdt};
Model sir;

int getNumSpecies() {
  return nS;
}

int getNumParams() {
  return nP;
}

const char** getNames() {
  return names;
}


Model getModel(double* initVals, double* params) {
    if (initVals) {
        sir.initVals = initVals;
    } else {
        sir.initVals = defInitVals;
    }

    if (params) {
        sir.params = params;
    } else {
        sir.params = defParams;
    }

    sir.funcs = funcs;
    sir.numSpecies = nS;

    return sir;
}

File: Python_Only/SIR.py
class SIR():
    """docstring for SIR"""
    params = {
    'b':0.5,
    'k':0.3333,
    }

    values = {
    'S': [0.9],
    'I': [0.1],
    'R': [0]
    }

    def __init__(self, params, values):
        self.funcs = {
        'S': self.dSdt,
        'I': self.dIdt,
        'R': self.dRdt
        }

        if params:
            self.params = params

        if values:
            self.values = values

    def dSdt(self, last):
        return -1*self.params['b']*last['S']*last['I']

    def dIdt(self, last):
        return (self.params['b']*last['S']*last['I'])-(self.params['k']*last['I'])

    def dRdt(self, last):
        return self.params['k']*last['I']
